lib/msf/core/handler/bind_aws_instance_connect.rb

```diff
@@ -0,0 +1,347 @@
+# -*- coding: binary -*-
+module Msf
+module Handler
+
+require 'aws-sdk-ec2instanceconnect'
+###
+#
+# This module implements the AWS InstanceConnect handler.  This means that
+# it will attempt to connect to a remote host through the AWS InstanceConnect pipe for 
+# a period of time (typically the duration of an exploit) to see if a the 
+# agent has started listening.
+#
+###
+module BindAwsInstanceConnect
+  include Msf::Handler
+  #
+  # Returns the handler specific string representation, in this case
+  # 'bind_tcp'.
+  #
+  def self.handler_type
+    return "bind_aws_instance_connect"
+  end
+
+  #
+  # Returns the connection oriented general handler type, in this case bind.
+  #
+  def self.general_handler_type
+    "bind"
+  end
+
+  # A string suitable for displaying to the user
+  #
+  # @return [String]
+  def human_name
+    "Bind AWS InstanceConnect"
+  end
+
+  #
+  # Initializes a bind handler and adds the options common to all bind
+  # payloads, such as local port.
+  #
+  def initialize(info = {})
+    super
+
+    register_options(
+      [
+        OptString.new('EC2_ID', [true, 'The EC2 ID of the instance ', '']),
+        OptString.new('REGION', [true, 'AWS region containing the instance', 'us-east-1']),
+        OptString.new('ACCESS_KEY_ID', [false, 'AWS access key', nil]),
+        OptString.new('SECRET_ACCESS_KEY', [false, 'AWS secret key', nil]),
+        OptString.new('INSTANCE_USER', [false, 'Username on the EC2 instance with which to log-in']),
+        OptString.new('INSTANCE_PORT', [false, 'Serial port index on the EC2 instance to use for the connection']),
+        OptString.new('ROLE_ARN', [false, 'AWS assumed role ARN', nil]),
+        OptString.new('ROLE_SID', [false, 'AWS assumed role session ID', nil]),
+      ], Msf::Handler::BindAwsInstanceConnect)
+
+    register_advanced_options(
+      [
+        OptString.new('PRIVATE_KEY', [
+          false,
+          'The string value of the private key that will be used. If you are using MSFConsole,
+          this value should be set as file:PRIVATE_KEY_PATH. OpenSSH, RSA, DSA, and ECDSA private keys are supported.'
+        ]),
+        OptString.new('KEY_PASS', [false, 'Passphrase for SSH private key(s)']),
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false])
+      ], Msf::Handler::BindAwsInstanceConnect)
+
+    self.listener_threads = []
+    self.conn_threads = []
+    self.listener_pairs   = {}
+  end
+
+  #
+  # Kills off the connection threads if there are any hanging around.
+  #
+  def cleanup_handler
+    # Kill any remaining handle_connection threads that might
+    # be hanging around
+    stop_handler
+    conn_threads.each { |thr|
+      thr.kill
```

Comment:
I'm surprised there isn't a try catch here:

```suggestion
      begin
        thr.kill
      rescue => e
        elog(e)
      end
```

Looks like we're inconsistent across modules though


https://github.com/rapid7/metasploit-framework/blob/6e5f528332da849f9a33284ca6b3f5443d64f176/lib/msf/core/handler/reverse_tcp_double.rb#L84-L88

https://github.com/rapid7/metasploit-framework/blob/6e5f528332da849f9a33284ca6b3f5443d64f176/lib/msf/core/handler/bind_tcp.rb#L55-L59